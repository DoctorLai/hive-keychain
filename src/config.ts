const Config = {
  hiveEngine: {
    mainNet: 'ssc-mainnet-hive',
  },
  claims: {
    FREQUENCY: +(process.env.DEV_CLAIM_FREQUENCY || 10 * 60 * 1000),
    freeAccount: {
      MIN_RC_PCT: +(process.env.DEV_CLAIM_ACCOUNT_RC_PCT || 95),
      MIN_RC: +(process.env.DEV_CLAIM_ACCOUNT_MIN_RC || 1.1 * 10 * 10 ** 12), // 10% more than 10^13 (current creation cost)
    },
  },
};

export default Config;
