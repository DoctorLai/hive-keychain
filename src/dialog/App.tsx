import { DialogCommand } from '@reference-data/dialog-message-key.enum';
import React, { useEffect, useState } from 'react';
import Unlock from 'src/dialog/pages/unlock';
import './dialog.scss';

const App = () => {
  useEffect(() => {
    chrome.runtime.onMessage.addListener(function (data, sender, sendResp) {
      setData(data);
    });
  }, []);

  const [data, setData] = useState<any>({});

  const renderDialogContent = () => {
    console.log(data);
    switch (data.command) {
      case DialogCommand.UNLOCK:
      case DialogCommand.WRONG_MK:
        return (
          <Unlock
            data={data}
            wrongMk={data.command === DialogCommand.WRONG_MK}
          />
        );
      default:
        return null;
    }
  };

  return <div className="dialog">{renderDialogContent()}</div>;
};

export default App;
