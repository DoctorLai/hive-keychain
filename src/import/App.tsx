import React, { Component } from 'react';
import { BackgroundCommand } from 'src/reference-data/background-message-key.enum';
import FileUtils from 'src/utils/file.utils';
import './import-keys.scss';

interface ImportAccountState {
  selectedFile?: File;
}

class App extends Component {
  state: ImportAccountState = {};

  handleFileUpload = (event: any) => {
    this.state.selectedFile = event.target.files[0];
  };

  importKeysFromFile = async () => {
    if (this.state.selectedFile) {
      const base64 = await FileUtils.toBase64(this.state.selectedFile);
      const fileData = atob(base64);
      chrome.runtime.sendMessage({
        command: BackgroundCommand.IMPORT_ACCOUNTS,
        value: fileData,
      });
    }
  };

  render() {
    return (
      <div className="import-keys">
        <div className="title-panel">
          <img src="/assets/images/iconhive.png" />
          <div className="title">
            {chrome.i18n.getMessage('import_html_title')}
          </div>
        </div>
        <div
          className="caption"
          dangerouslySetInnerHTML={{
            __html: chrome.i18n.getMessage('import_html_text'),
          }}></div>

        <input
          type="file"
          accept=".kc"
          id="file"
          onChange={this.handleFileUpload}
        />
        <label htmlFor="file">Choose a file</label>
        <span id="file_span">{this.state.selectedFile?.name}</span>
        <button id="proceed" onClick={this.importKeysFromFile}>
          {chrome.i18n.getMessage('popup_html_import')}
        </button>
      </div>
    );
  }
}

export default App;
