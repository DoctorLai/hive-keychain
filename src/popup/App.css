.App {
    height: 600px;
    width: 300px;
    font-family: Futura !important;
    font-weight: 400;
    background: url("/assets/images/hive-watermark.png") 0% 0% no-repeat
    padding-box, linear-gradient(330deg, #59626c 0%, #000000 100%) 0%
    0% no-repeat padding-box !important;

    color: white;

}