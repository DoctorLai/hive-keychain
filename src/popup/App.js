import React, { Component } from "react";

class App extends Component {
  render() {
    return (
      <div className="App">
        <h1>{chrome.i18n.getMessage("popup_html_new_password")}</h1>
      </div>
    );
  }
}

export default App;
