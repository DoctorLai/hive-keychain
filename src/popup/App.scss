@import './variable.scss';
@font-face {
  font-family: Futura;
  src: url('/assets/fonts/Futura-Boo.otf');
  font-weight: 400;
}

.App {
  height: 600px;
  width: 350px;
  font-family: Futura !important;
  font-weight: 400;
  background: url('/assets/images/hive-watermark.png') 0% 0% no-repeat
      padding-box,
    linear-gradient(330deg, #59626c 0%, #000000 100%) 0% 0% no-repeat
      padding-box !important;
  color: white;
  box-sizing: border-box;
  overflow: auto;

  &.homepage {
    background: url(/assets/images/hive-watermark.png) 0% 0% no-repeat
        padding-box,
      linear-gradient(to bottom, #000000, #59626c, #000000) 0% 0% no-repeat
        padding-box !important;
  }

  .page-container {
    display: grid;
    row-gap: 32px;
    grid-template-rows: 28px 1fr;
    height: 100%;
  }
  .divider {
    flex: 1;
  }
}

.react-dropdown-select-dropdown-handle {
  width: 48px;
  svg {
    width: 20px !important;
    height: 20px !important;
  }
}

.pretty.p-switch .state:before {
  top: calc((0% - (100% - 1em)) - 13%) !important;
}

.css-wmw4vi-ReactDropdownSelect {
  padding: 2px 0px !important;
}

.Toastify__progress-bar {
  bottom: unset;
  top: 0;
}

.Toastify__toast-icon {
  display: none !important;
}

.Toastify__progress-bar {
  background: white !important;
}

.Toastify__toast--error {
  background: $message-container-error !important;
}
.Toastify__toast--info {
  background: $message-container-info !important;
}
.Toastify__toast--success {
  background: $message-container-success !important;
}
