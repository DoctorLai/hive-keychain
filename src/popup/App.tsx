import {retrieveAccounts} from '@popup/actions/account.actions';
import {setMk} from '@popup/actions/mk.actions';
import {navigateTo} from '@popup/actions/navigation.actions';
import {RootState} from '@popup/store';
import React, {useEffect} from 'react';
import {connect, ConnectedProps} from 'react-redux';
import {BackgroundMessage} from 'src/background/background-message.interface';
import {BackgroundCommand} from 'src/reference-data/background-message-key.enum';
import {Screen} from 'src/reference-data/screen.enum';
import PopupUtils from 'src/utils/popup.utils';
import './App.css';
import {AddAccountRouterComponent} from './pages/add-account/add-account-router/add-account-router.component';
import {AppContainerComponent} from './pages/app-container/app-container.component';
import {ErrorMessageContainerComponent} from './pages/error-message-container/error-message-container.component';
import {SignInComponent} from './pages/sign-in/sign-in.component';
import {SignUpComponent} from './pages/sign-up/sign-up.component';

const App = ({
  setMk,
  mk,
  retrieveAccounts,
  accounts,
  navigateTo,
}: PropsFromRedux) => {
  useEffect(() => {
    PopupUtils.fixPopupOnMacOs();
  }, []);

  useEffect(() => {
    chrome.runtime.sendMessage({command: BackgroundCommand.GET_MK});
    chrome.runtime.onMessage.addListener(onSentBackMkListener);
  }, [setMk]);

  const onSentBackMkListener = (message: BackgroundMessage) => {
    if (message.command === BackgroundCommand.SEND_BACK_MK) {
      setMk(message.value);
      retrieveAccounts(message.value);
      chrome.runtime.onMessage.removeListener(onSentBackMkListener);
    }
  };

  const renderMainLayoutNav = () => {
    if (!mk) {
      if (accounts.length === 0) {
        return <SignUpComponent />;
      } else {
        return <SignInComponent />;
      }
    } else {
      if (accounts.length === 0) {
        navigateTo(Screen.ACCOUNT_PAGE_INIT_ACCOUNT);
        return <AddAccountRouterComponent />;
      } else {
        return <AppContainerComponent />;
      }
    }
  };

  return (
    <div className="App">
      {renderMainLayoutNav()}
      <ErrorMessageContainerComponent />
    </div>
  );
};

const mapStateToProps = (state: RootState) => {
  return {
    mk: state.mk,
    accounts: state.accounts,
  };
};

const connector = connect(mapStateToProps, {
  setMk,
  retrieveAccounts,
  navigateTo,
});
type PropsFromRedux = ConnectedProps<typeof connector>;

export default connector(App);
