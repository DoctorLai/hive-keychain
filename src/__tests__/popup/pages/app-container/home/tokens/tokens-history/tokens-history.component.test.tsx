import App from '@popup/App';
import { screen } from '@testing-library/react';
import React from 'react';
import tokensHistory from 'src/__tests__/popup/pages/app-container/home/tokens/tokens-history/mocks/tokens-history';
import config from 'src/__tests__/utils-for-testing/setups/config';
config.byDefault();
const { methods } = tokensHistory;
describe('tokens-history.component tests:\n', () => {
  methods.afterEach;
  describe('With history to show', () => {
    beforeEach(async () => {
      await tokensHistory.beforeEach(<App />);
    });
    it('Must show LEO token history and equal snapshot', () => {
      screen.debug();
    });
  });
  describe('No history to show', () => {
    it.todo('Must show no transactions');
  });
});
