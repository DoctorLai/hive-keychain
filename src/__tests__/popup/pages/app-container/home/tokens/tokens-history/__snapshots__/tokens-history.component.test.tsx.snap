// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokens-history.component tests:
 No history to show Must show no transactions: tokens-history.component NO DATA 1`] = `
<DocumentFragment>
  <div
    class="App "
  >
    <div
      class="app-router"
      style="height: 100%; display: grid; grid-template-rows: 70px 1fr;"
    >
      <div
        class="title-section"
      >
        <span
          aria-label="arrow-back-icon"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          arrow_back
        </span>
        <div
          class="title"
        >
          LEO history
        </div>
        <span
          aria-label="icon-close-page"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          clear
        </span>
      </div>
      <div
        class="page-content"
        style="overflow: auto; display: flex; flex-direction: column;"
      >
        <div
          aria-label="tokens-history"
          class="tokens-history"
        >
          <div
            class="custom-input"
          >
            <div
              class="input-container no-logo   "
            >
              <input
                aria-label="input-filter-box"
                class=""
                placeholder="Search "
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="item-list"
          />
        </div>
      </div>
    </div>
    <div
      class="Toastify"
    />
    <div
      aria-label="proposal-voting-section"
      class="proposal-voting-section hide closed"
    >
      <span
        class="icon-component material-icons close clickable"
      >
        clear
      </span>
      <div
        class="text"
      >
        Support Keychain development proposal
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`tokens-history.component tests:
 With history to show Must show LEO token history and equal snapshot: tokens-history.component 1`] = `
<DocumentFragment>
  <div
    class="App "
  >
    <div
      class="app-router"
      style="height: 100%; display: grid; grid-template-rows: 70px 1fr;"
    >
      <div
        class="title-section"
      >
        <span
          aria-label="arrow-back-icon"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          arrow_back
        </span>
        <div
          class="title"
        >
          LEO history
        </div>
        <span
          aria-label="icon-close-page"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          clear
        </span>
      </div>
      <div
        class="page-content"
        style="overflow: auto; display: flex; flex-direction: column;"
      >
        <div
          aria-label="tokens-history"
          class="tokens-history"
        >
          <div
            class="custom-input"
          >
            <div
              class="input-container no-logo   "
            >
              <input
                aria-label="input-filter-box"
                class=""
                placeholder="Search "
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="item-list"
          >
            <div
              aria-label="token-history-item-LEO"
              class="token-history-item has-memo"
              id="61674a248bae1252026e04ef"
            >
              <div
                class="transaction"
              >
                <div
                  class="information-panel"
                >
                  <div
                    class="top-row"
                  >
                    <div
                      class="date"
                    >
                      10/13/2021
                    </div>
                  </div>
                  <div
                    class="bottom-row"
                  >
                    Received 0.986 LEO as comment curation reward
                  </div>
                </div>
                <div
                  class="memo-panel closed"
                >
                  re-theghost1980-20211215t195955670z
                </div>
              </div>
            </div>
            <div
              aria-label="token-history-item-LEO"
              class="token-history-item has-memo"
              id="61654d8f8bae125202648bbd"
            >
              <div
                class="transaction"
              >
                <div
                  class="information-panel"
                >
                  <div
                    class="top-row"
                  >
                    <div
                      class="date"
                    >
                      10/12/2021
                    </div>
                  </div>
                  <div
                    class="bottom-row"
                  >
                    Received 0.985 LEO as author reward
                  </div>
                </div>
                <div
                  class="memo-panel closed"
                >
                  re-theghost1980-20211215t195955670z
                </div>
              </div>
            </div>
            <div
              aria-label="token-history-item-LEO"
              class="token-history-item "
              id="615a3b378bae1252023114ba"
            >
              <div
                class="transaction"
              >
                <div
                  class="information-panel"
                >
                  <div
                    class="top-row"
                  >
                    <div
                      class="date"
                    >
                      10/03/2021
                    </div>
                  </div>
                  <div
                    class="bottom-row"
                  >
                    Received 10000.985 LEO from mining lottery (10099)
                  </div>
                </div>
                <div
                  class="memo-panel closed"
                />
              </div>
            </div>
            <div
              aria-label="token-history-item-LEO"
              class="token-history-item has-memo"
              id="6147d1928bae125202dd666e"
            >
              <div
                class="transaction"
              >
                <div
                  class="information-panel"
                >
                  <div
                    class="top-row"
                  >
                    <div
                      class="date"
                    >
                      09/19/2021
                    </div>
                  </div>
                  <div
                    class="bottom-row"
                  >
                    Received 0.985 LEO from @contract_tokens
                  </div>
                </div>
                <div
                  class="memo-panel closed"
                >
                  Fake Memo tokens_transfer
                </div>
              </div>
            </div>
            <div
              aria-label="token-history-item-LEO"
              class="token-history-item "
              id="613ef535780fd34f81d69aef"
            >
              <div
                class="transaction"
              >
                <div
                  class="information-panel"
                >
                  <div
                    class="top-row"
                  >
                    <div
                      class="date"
                    >
                      09/10/2021
                    </div>
                  </div>
                  <div
                    class="bottom-row"
                  >
                    @contract_tokens staked 1000.986 LEO for @keychain.tests
                  </div>
                </div>
                <div
                  class="memo-panel closed"
                />
              </div>
            </div>
            <div
              aria-label="token-history-item-LEO"
              class="token-history-item "
              id="6135cb1c187544f7ef365d21"
            >
              <div
                class="transaction"
              >
                <div
                  class="information-panel"
                >
                  <div
                    class="top-row"
                  >
                    <div
                      class="date"
                    >
                      08/21/2021
                    </div>
                  </div>
                  <div
                    class="bottom-row"
                  >
                    Delegated 100000.985 LEO to @theghost1980
                  </div>
                </div>
                <div
                  class="memo-panel closed"
                />
              </div>
            </div>
            <div
              aria-label="token-history-item-LEO"
              class="token-history-item "
              id="6135ca75187544f7ef2f3833"
            >
              <div
                class="transaction"
              >
                <div
                  class="information-panel"
                >
                  <div
                    class="top-row"
                  >
                    <div
                      class="date"
                    >
                      08/20/2021
                    </div>
                  </div>
                  <div
                    class="bottom-row"
                  >
                    Canceled a 0.985 LEO delegation to @contract_tokens
                  </div>
                </div>
                <div
                  class="memo-panel closed"
                />
              </div>
            </div>
            <div
              aria-label="token-history-item-LEO"
              class="token-history-item "
              id="6135999e187544f7efe79c78"
            >
              <div
                class="transaction"
              >
                <div
                  class="information-panel"
                >
                  <div
                    class="top-row"
                  >
                    <div
                      class="date"
                    >
                      05/08/2021
                    </div>
                  </div>
                  <div
                    class="bottom-row"
                  >
                    Canceled a delegation of 6.666 LEO LEO to @contract_tokens
                  </div>
                </div>
                <div
                  class="memo-panel closed"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Toastify"
    />
    <div
      aria-label="proposal-voting-section"
      class="proposal-voting-section hide closed"
    >
      <span
        class="icon-component material-icons close clickable"
      >
        clear
      </span>
      <div
        class="text"
      >
        Support Keychain development proposal
      </div>
    </div>
  </div>
</DocumentFragment>
`;
